File: app/src/main/java/com/omar/retromp3recorder/app/files/FilePathGeneratorImpl.java
package com.omar.retromp3recorder.app.files;

import android.content.Context;

import com.omar.retromp3recorder.app.utils.NotUnitTestable;

import java.io.File;

import javax.inject.Inject;

@NotUnitTestable
public class FilePathGeneratorImpl implements FilePathGenerator {
    private static final String MP3_EXTENSION = ".mp3";
    private static final String VOICE_RECORD = "voice_record";

    private final Context context;

     public FilePathGeneratorImpl(Context context) {
        this.context = context;
    }

    @Override
    public String generateFilePath() {
        String fileName = VOICE_RECORD + "_" + String.valueOf(System.currentTimeMillis()) + MP3_EXTENSION;
        return getFileDir() + "/" + fileName;
    }

    @Override
    public String getFileDir() {
        return context.getExternalCacheDir().toString();
    }
}

File: io-files/src/main/java/com/omar/retromp3recorder/files/FilePathGeneratorImpl.kt
package com.omar.retromp3recorder.files

import android.content.Context
import com.omar.retromp3recorder.utils.NotUnitTestable

@NotUnitTestable
class FilePathGeneratorImpl(
    private val context: Context
) : FilePathGenerator {
    override fun generateFilePath(): String {
        val fileName = VOICE_RECORD + "_" + System.currentTimeMillis() + MP3_EXTENSION
        return "$fileDir/$fileName"
    }

    override val fileDir: String
        get() = context.externalCacheDir.toString()
}

private const val MP3_EXTENSION = ".mp3"
private const val VOICE_RECORD = "voice_record"